servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/DefaultAsyncContextProviderTest.java

```diff
@@ -468,6 +473,236 @@ void testContextInPublisherOperators() throws Exception {
         // TODO we don't have nested operators such as flatMap yet, to be tested when we implement them
     }
 
+    private enum WrapType {
+        NONE,
+        SEQUENTIAL,
+        CONCURRENT,
+    }
+
+    @ParameterizedTest(name = "{displayName} [{index}] wrapType={0}")
+    @EnumSource(WrapType.class)
+    void multipleCompletableWrapSubscriber(WrapType wrapType) throws Exception {
+        CompletableFuture<CapturedContext> f1 = new CompletableFuture<>();
+        CompletableSource.Subscriber subscriber = new CompletableSource.Subscriber() {
+            @Override
+            public void onSubscribe(Cancellable cancellable) {
+            }
+
+            @Override
+            public void onComplete() {
+                f1.complete(AsyncContext.captureContext());
+            }
+
+            @Override
+            public void onError(Throwable t) {
+            }
+        };
+        CapturedContext c1 = new CopyOnWriteContextMap();
+        CapturedContext c2 = new CopyOnWriteContextMap();
+        CompletableSource.Subscriber wrapped = INSTANCE.wrapCompletableSubscriber(subscriber, c1);
+        switch (wrapType) {
+            case SEQUENTIAL:
+                wrapped = INSTANCE.wrapCancellable(wrapped, c2);
+                wrapped = INSTANCE.wrapCompletableSubscriber(wrapped, c2);
+                break;
+            case CONCURRENT:
+                wrapped = INSTANCE.wrapCompletableSubscriberAndCancellable(wrapped, c2);
+                break;
+            default: // NONE
```

Comment:
There is a risk that if someone adds another enum value, behavior might be unpredictable or hard to debug. Consider using `case NONE` explicitly and then implement `default` as:
```java
throw new IllegalArgumentException("Unsupported WrapType: " + wrapType);
```